### PR TITLE
OGM-1287 Create a property to look up native nosql connection via JNDI

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/OGM-1287

### CODE REVIEW

File: documentation/manual/src/main/asciidoc/modules/configuration.asciidoc [hunk 1]
```diff
@@ -606,4 +606,105 @@ Then similarly to what you did with the Hibernate OGM modules zip, unpack this o
...
+</subsystem>
+----
+
+Notice that here the *jndi-name* attribute defines the String for the external lookup,
+it will be used later in this chapter.
+While *module* attribute indicates the static module from which to load client driver.
+
+[NOTE]
+====
+If you have provisioned your WildFly with Hibernate Ogm featurepack(s),
```
Comment (Sanne) [inline documentation/manual/src/main/asciidoc/modules/configuration.asciidoc]: s/Ogm/OGM/

File: documentation/manual/src/main/asciidoc/modules/configuration.asciidoc [hunk 2]
```diff
@@ -606,4 +606,105 @@ Then similarly to what you did with the Hibernate OGM modules zip, unpack this o
...
+[subs="verbatim,attributes"]
+----
+<socket-binding-group ...
+	<outbound-socket-binding name="mongodb">
+		<remote-destination host="localhost" port="27018"/>
+	</outbound-socket-binding>
+</socket-binding-group>
+----
+
+At this point you can use Hibernate property `hibernate.connection.resource` in your _persistence.xml_,
```
Comment (Sanne) [inline documentation/manual/src/main/asciidoc/modules/configuration.asciidoc]: s/can use Hibernate property/can use the Hibernate property/

**Comment by fax4ever** [comment]
Thank you @Sanne for the typo fixes. I applied them and rebased the branch.

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java [hunk 3]
```diff
@@ -109,6 +112,28 @@ public boolean allowsTransactionEmulation() {
...
+		try {
+			log.tracef( "Retrieving MongoDatabase from JNDI at %1$s", config.getNativeClientResource() );
+			mongoDb = (MongoDatabase) jndiService.locate( config.getNativeClientResource() );
+		}
+		catch (RuntimeException e) {
+			throw log.errorOnFetchJndiClientProperty( config.getNativeClientResource() );
+		}
+
+		// clear resources
+		this.jndiService = null;
```
Comment (DavideD) [inline mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java]: Not critical but I find it counter intuitive that the lookup method also does the clear, can we move it at the end of the start method?

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jDatastoreProvider.java [hunk 4]
```diff
@@ -52,17 +58,42 @@ public TransactionCoordinatorBuilder getTransactionCoordinatorBuilder(Transactio
...
+			log.tracef( "Retrieving MongoDatabase from JNDI at %1$s", configuration.getNativeClientResource() );
+			Driver driver = (Driver) jndiService.locate( configuration.getNativeClientResource() );
+			this.client = new BoltNeo4jClient( driver );
+		}
+		catch (RuntimeException e) {
+			throw log.errorOnFetchJndiClientProperty( configuration.getNativeClientResource() );
+		}
+
+		// clear resources
+		this.jndiService = null;
```
Comment (DavideD) [inline neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jDatastoreProvider.java]: Not critical but I find it counter intuitive that the lookup method also does the clear, can we move it at the end of the start method?

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java (hunk 3 above)
Comment (fax4ever) [reply mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java]: Sure I can

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jDatastoreProvider.java (hunk 4 above)
Comment (fax4ever) [reply neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jDatastoreProvider.java]: Sure I can

**Comment by DavideD** [comment]
> would it be possible to add some tests in the integrationtest module?

Are you working on this?

**Comment by fax4ever** [comment]
@DavideD good idea. I'm going to start to work on it.

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-ogm/pull/1045

Thanks @fax4ever!
I had to change a commit message because the issue was wrong and fix a minor thing in the documentation

**Comment by fax4ever** [comment]
Thank you @DavideD

**Pull request closed without merging**